govc/library/vmtx_item_info.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vapi/library"
+	"github.com/vmware/govmomi/vapi/vcenter"
+)
+
+type vmtxItemInfo struct {
+	*flags.ClientFlag
+	*flags.OutputFlag
+}
+
+type vmtxItemInfoResultsWriter struct {
+	Result *vcenter.TemplateInfo
+	m      *vcenter.Manager
+	cmd    *vmtxItemInfo
+}
+
+func (r vmtxItemInfoResultsWriter) MarshalJSON() ([]byte, error) {
+	return json.Marshal(r.Result)
+}
+
+func (r vmtxItemInfoResultsWriter) Write(w io.Writer) error {
+	tw := tabwriter.NewWriter(w, 2, 0, 2, ' ', 0)
+	defer tw.Flush()
+	if err := r.writeLibraryTemplateDetails(tw, *r.Result); err != nil {
+		return err
+	}
+	tw.Flush()
+	return nil
+}
+
+func (r vmtxItemInfoResultsWriter) writeLibraryTemplateDetails(
+	w io.Writer, v vcenter.TemplateInfo) error {
+
+	fmt.Fprintf(w, "  VM Template:\t%s\n", v.VmTemplate)
+	fmt.Fprintf(w, "  Guest OS:\t%s\n", v.GuestOS)
+	fmt.Fprintf(w, "  CPU:\t\n")
+	fmt.Fprintf(w, "    Count:\t%d\n", v.CPU.Count)
+	fmt.Fprintf(w, "    Cores Per Socket:\t%d\n", v.CPU.CoresPerSocket)
+	fmt.Fprintf(w, "  Memory:\t\n")
+	fmt.Fprintf(w, "    Size in MB:\t%d\n", v.Memory.SizeMB)
+	fmt.Fprintf(w, "  Disks:\t\n")
+	for _, d := range v.Disks {
+		fmt.Fprintf(w, "    Key:\t%s\n", d.Key)
+		fmt.Fprintf(w, "    Capacity:\t%d\n", d.Value.Capacity)
+		fmt.Fprintf(w, "    Datastroe:\t%s\n\n", d.Value.DiskStorage.Datastore)
```

Comment:
question: none of these (sub)fields use pointer semantics, i.e. no risk of NPE correct?

Reply:
Yes

Reply:
Done